pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTLiteral.java

```diff
@@ -252,6 +258,100 @@ public boolean isSingleCharacterStringLiteral() {
     }
 
     public boolean isTextBlock() {
-        return isString && getImage().startsWith("\"\"\"");
+        return isString && getImage().startsWith(TEXTBLOCK_DELIMITER);
+    }
+
+    /**
+     * Returns the content of the text block after normalizing line endings to LF,
+     * removing incidental white space surrounding the text block and interpreting
+     * escape sequences.
+     */
+    @Experimental
+    public String getTextBlockContent() {
+        if (!isTextBlock()) {
+            return getImage();
+        }
+
+        int start = determineContentStart(getImage());
+        String content = getImage().substring(start, getImage().length() - TEXTBLOCK_DELIMITER.length());
+        // normalize line endings to LF
+        content = content.replaceAll("\r\n|\r", "\n");
+
+        int prefixLength = Integer.MAX_VALUE;
+        List<String> lines = Arrays.asList(content.split("\\n"));
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            // compute common prefix
+            if (!StringUtils.isAllBlank(line) || i == lines.size() - 1) {
+                prefixLength = Math.min(prefixLength, countLeadingWhitespace(line));
+            }
+        }
+        if (prefixLength == Integer.MAX_VALUE) {
+            // common prefix not found
+            prefixLength = 0;
+        }
+        StringBuilder sb = new StringBuilder(content.length());
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            // remove common whitespace prefix
+            if (!StringUtils.isAllBlank(line) && line.length() >= prefixLength) {
+                line = line.substring(prefixLength);
+            }
+            line = removeTrailingWhitespace(line);
+            sb.append(line);
+
+            boolean isLastLine = i == lines.size() - 1;
+            boolean isFirstLine = i == 0;
+            if (!isLastLine || !isFirstLine && !StringUtils.isAllBlank(line)) {
+                sb.append('\n');
+            }
+        }
+        String result = sb.toString();
+
+        // interpret escape sequences "\NL" "n","t","b","r","f", "s", "\"", "\'"
```

Comment:
I think this is missing `\\` for a backslash

Reply:
Good catch!